doc/SDK_NOTIFICATIONS.md

```diff
@@ -0,0 +1,77 @@
+ - When creating an on-chain receive swap address - Use breez server API to register for notification of any tx associated with this address
+ - Use LSP API to register webhook to be notified when there is an incoming paypment(htlc) for the user node.
+
+### LSP
+- `subscribe_notifications(url, signature)` API for registering a webhook by sdk clients. The signature is the url signed by the node id. The LSP should maintain a mapping between node ids to web hooks URLs. These webhooks (from the client perspective) tend to change frequently as for example in push notifications a new token is generated from time to time so the client is responsible to update the url on any change.
+
+When an htlc is intercepted in the LSP and the destination peer is not online the LSP will notify the web hook with the right type and data, signling incoming payment, in attempt to wake up the destination node, wait for a some time (how much?) and once the peer is online will forward the htlcs.
+
+### Chain notification service
+Breez server will implement the chain notifications for sdk clients. Connected to bitcoind breez server download every block, check for webhook registration and trigger notifications. API key is needed to subscribe for such notification.
+The following endpoints are needed:
+- `registration_id add_registration(current_registration_id, url)` - get the current registration_id if exists or create one if the former was deleted. Also associate the url with the new registration if is not already associated. At the first time this endpoint is called the `current_registration_id` can be null. This mechanism allows the service to delete expired registrations according to its own policy requiring the client to refresh the registration from time to time signaling it is truly active.
+- `subscribe_address_transactions(registration_id, url, address)` API for subscribing to address notifications.
+- `subscribe_tx_confirmation(registration_id, url, txid)` API for subscribing to specific transaction notifications.
+The `registration_id` is a random unique identifier that identify the caller and allow the sdk client to both update the hook url and subscribe to notification services.
+
+### Notification Delivery Service
+This service is the destination of the web hooks. Its job is to identify the protocol, type and data of the notification, craft the formatted notification and send it over the right channel (mobile push for example)
+Such service should be opertated by an app provider that would like to send push notifications to users.
+
+## Webhook structure
+A webhook is just a url but for mobile we require `platform` and `token` parameters for example: <base_url>?platform=<platform>&token=<token>
+Where:
+- `platform` is one of: apn, fcm
+- `token` is the token given by the client OS.
+
+Triggering the webhook is done by initiating a POST request with the following json payload:
+
+{
+ "type": <hook_type>,
+ "data": <hook_data>
+}
```

Comment:
a timestamp is usually good practice to add in webhook contents too.
And I'm thinking a signature could be beneficial, because the webhook endpoint will be a rather open endpoint with credentials in the url. (you want to somehow restrict the services that can call the webhook endpoint in order to wake up your mobile phone)